modules/order/src/EventSubscriber/OrderNumberSubscriber.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Drupal\commerce_order\EventSubscriber;
+
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class OrderNumberSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events = [
+      'commerce_order.place.pre_transition' => ['setOrderNumber', -100],
```

Comment:
We'll want to document this, considering it's on event level priority and not resolving like our normal cases.